libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt

```diff
@@ -567,17 +577,12 @@ constructor(
     }
 
     private fun reroute() {
-        ifNonNull(tripSession.getEnhancedLocation()) { location ->
-            val optionsRebuilt = AdjustedRouteOptionsProvider.getRouteOptions(
-                directionsSession,
-                tripSession,
-                location
-            ) ?: return
-            directionsSession.requestRoutes(
-                optionsRebuilt,
-                null
-            )
-        }
+        val optionsRebuilt = routingOptionsProvider.provideOffRouteRefreshAdapter().newRouteOptions(
+            directionsSession.getRouteOptions(),
+            tripSession.getRouteProgress(),
+            tripSession.getEnhancedLocation()
+        ) ?: return
```

Comment:
2 nits: 
1. remove provideOffRouteRefreshAdapter
1. remove ?: return in favor of kotlin ?. operation (such as let)

``` kotlin
val newRouteOptions = routingOptionsProvider?.newRouteOptions(
    directionsSession.getRouteOptions(),
    tripSession.getRouteProgress(),
    tripSession.getEnhancedLocation())
newRouteOptions?.let {
   directionsSession.requestRoutes(newRouteOptions, null)
}
```